contracts/root/predicates/ERC20Predicate.sol

```diff
@@ -0,0 +1,191 @@
+
+  constructor(address _withdrawManager) public IPredicate(_withdrawManager) {}
+
+  /**
+   * @notice Start an exit from the side chain by referencing the preceding (reference) transaction
+   * @param data RLP encoded data of the reference tx that encodes the following fields:
+   * headerNumber Header block number of which the reference tx was a part of
+   * blockProof Proof that the block header (in the child chain) is a leaf in the submitted merkle root
+   * blockNumber Block number of which the reference tx is a part of
+   * blockTime Reference tx block time
+   * blocktxRoot Transactions root of block
+   * blockReceiptsRoot Receipts root of block
+   * receipt Receipt of the reference transaction
+   * receiptProof Merkle proof of the reference receipt
+   * branchMask Merkle proof branchMask for the receipt
+   * logIndex Log Index to read from the receipt
+   * exitTx Signed exit transaction
+   */
+  function startExit(bytes memory data)
+    public
+  {
+    RLPReader.RLPItem[] memory referenceTxData = data.toRlpItem().toList();
+    uint256 age = withdrawManager.verifyInclusion(data);
+    // validate exitTx
+    uint256 exitAmountOrTokenId;
+    address childToken;
+    address participant;
+    bool burnt;
+    (exitAmountOrTokenId, childToken, participant, burnt) = processExitTx(referenceTxData[10].toBytes());
+
+    // process the receipt of the referenced tx
+    address rootToken;
+    uint256 oIndex;
+    (rootToken, oIndex) = processReferenceTx(
+      referenceTxData[6].toBytes(), // receipt
+      referenceTxData[9].toUint(), // logIndex
+      participant,
+      childToken,
+      exitAmountOrTokenId
+    );
+    withdrawManager.addExitToQueue(msg.sender, childToken, rootToken, exitAmountOrTokenId, burnt, age + oIndex);
+  }
+
+  /**
+   * @notice Process the reference tx to start a MoreVP style exit
+   * @param receipt Receipt of the reference transaction
+   * @param logIndex Log Index to read from the receipt
+   * @param participant Either of exitor or a counterparty depending on the type of exit
+   */
+  function processReferenceTx(
+    bytes memory receipt,
+    uint256 logIndex,
+    address participant,
+    address childToken,
+    uint256 exitAmountOrTokenId)
+    public
+    view
+    returns(address rootToken, uint256 oIndex)
+  {
+    RLPReader.RLPItem[] memory inputItems = receipt.toRlpItem().toList();
+    inputItems = inputItems[3].toList()[logIndex].toList(); // select log based on given logIndex
+    require(
+      childToken == RLPReader.toAddress(inputItems[0]), // "address" (contract address that emitted the log) field in the receipt
+      "Reference and exit tx do not correspond to the same token"
+    );
+    bytes memory logData = inputItems[2].toBytes();
+    inputItems = inputItems[1].toList(); // topics
+    // now, inputItems[i] refers to i-th (0-based) topic in the topics array
+    // inputItems[0] is the event signature
+    rootToken = address(RLPReader.toUint(inputItems[1]));
```

Comment:
Please fix `toAddress` issue

Reply:
Will fix all these nits in a single PR together.